tests/unit/anchore_engine/services/policy_engine/engine/feeds/test_grypedb_sync.py

```diff
@@ -0,0 +1,174 @@
+        """
+
+        def _mock_query(mocked_output):
+            mock_active_dbs = Mock(return_value=mocked_output)
+            GrypeDBSyncManager._query_active_dbs = mock_active_dbs
+
+        return _mock_query
+
+    @pytest.fixture
+    def mock_get_local_grypedb_checksum(self):
+        """
+        Creates factory that will mock _query_active_dbs return with params passed
+        """
+
+        def _mock_query(mocked_output):
+            mock_checksum = Mock(return_value=mocked_output)
+            GrypeDBSyncManager.get_local_grypedb_checksum = mock_checksum
+
+        return _mock_query
+
+    @pytest.fixture
+    def mock_calls_for_sync(
+        self, mock_query_active_dbs_with_data, mock_get_local_grypedb_checksum
+    ):
+        """
+        Provides ability to mock all class methods necessary to run a sync as a unit test
+        """
+
+        def _mock(mock_active_dbs=[], mock_local_checksum=""):
+            mock_query_active_dbs_with_data(mock_active_dbs)
+            mock_get_local_grypedb_checksum(mock_local_checksum)
+
+        return _mock
+
+    def test_no_active_grypedb(self, mock_calls_for_sync):
+        mock_calls_for_sync(
+            mock_active_dbs=[],
+            mock_local_checksum="eef3b1bcd5728346cb1b30eae09647348bacfbde3ba225d70cb0374da249277c",
+        )
+        result = GrypeDBSyncManager.run_grypedb_sync()
+
+        assert not result
+
+    def test_too_many_active_grypedbs(self, mock_query_active_dbs_with_data):
+        active_dbs = [GrypeDBMetadata(active=True), GrypeDBMetadata(active=True)]
+        mock_query_active_dbs_with_data(active_dbs)
+
+        with pytest.raises(TooManyActiveGrypeDBs):
+            GrypeDBSyncManager.run_grypedb_sync()
+
+    def test_matching_checksums(self, mock_calls_for_sync):
+        checksum = "eef3b1bcd5728346cb1b30eae09647348bacfbde3ba225d70cb0374da249277c"
+        mock_calls_for_sync(
+            mock_active_dbs=[GrypeDBMetadata(checksum=checksum)],
+            mock_local_checksum=checksum,
+        )
+
+        sync_ran = GrypeDBSyncManager.run_grypedb_sync()
+
+        assert not sync_ran
+
+    def test_mismatch_checksum(self, mock_calls_for_sync):
+        global_checksum = (
+            "eef3b1bcd5728346cb1b30eae09647348bacfbde3ba225d70cb0374da249277c"
+        )
+        local_checksum = (
+            "366ab0a94f4ed9c22f5cc93e4d8f6724163a357ae5190740c1b5f251fd706cc4"
+        )
+
+        mock_calls_for_sync(
+            mock_active_dbs=[GrypeDBMetadata(checksum=global_checksum)],
+            mock_local_checksum=local_checksum,
+        )
+
+        # mock execution of update
+        GrypeDBSyncManager.update_grypedb = Mock(return_value=True)
+
+        # pass a file path to bypass connection to catalog to retrieve tar from object storage
+        sync_ran = GrypeDBSyncManager.run_grypedb_sync(
+            grypedb_file_path="test/bypass/catalog.txt"
+        )
+
+        assert sync_ran
```

Comment:
`sync_ran` should be.... `True` ? If so, then it is better to assert what this is expected to be:
```suggestion
        assert sync_ran is True
```